src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,829 @@
+            var generator = SyntaxGenerator.GetGenerator(document);
+            var semanticFacts = document.GetRequiredLanguageService<ISemanticFactsService>();
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var editor = new SyntaxEditor(root, generator);
+            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+            var insertionIndex = GetInsertionIndex(compilation, methodSymbol, syntaxFacts, containingMethod);
+
+            foreach (var invocationExpression in invocations)
+            {
+                var newArgumentExpression = expression;
+                var invocationArguments = syntaxFacts.GetArgumentsOfInvocationExpression(invocationExpression);
+                var variablesInExpression = expressionCopy.DescendantNodes().OfType<TIdentifierNameSyntax>();
+                var invocationSemanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+                editor.ReplaceNode(invocationExpression, (currentInvo, _) =>
+                {
+                    var updatedInvocationArguments = syntaxFacts.GetArgumentsOfInvocationExpression(currentInvo);
+                    var parameterIsNamed = false;
+
+                    foreach (var variable in variablesInExpression)
+                    {
+                        (newArgumentExpression, parameterIsNamed) = CreateNewArgumentExpression(
+                            invocationSemanticModel, document, syntaxFacts, mappingDictionary, generator,
+                            newArgumentExpression, variable, invocationArguments, updatedInvocationArguments,
+                            parameterIsNamed, cancellationToken);
+                    }
+
+                    var expressionFromInvocation = syntaxFacts.GetExpressionOfInvocationExpression(invocationExpression);
+                    var allArguments = AddArgumentToArgumentList(updatedInvocationArguments, generator,
+                        newArgumentExpression.WithAdditionalAnnotations(Formatter.Annotation), insertionIndex, parameterName, parameterIsNamed);
+                    var newInvo = editor.Generator.InvocationExpression(expressionFromInvocation, allArguments);
+                    return newInvo;
+                });
+            }
+
+            // If you are at the original document, then also introduce the new method and introduce the parameter.
+            if (document.Id == originalDocument.Id)
+            {
+                await UpdateExpressionInOriginalFunctionAsync(originalDocument, expressionCopy, containingMethod,
+                    parameterName, editor, allOccurrences, cancellationToken).ConfigureAwait(false);
+                var semanticModel = await originalDocument.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var parameterType = semanticModel.GetTypeInfo(expressionCopy, cancellationToken).Type ??
```

Comment:
why .Type and not .COnvertedTYpe?

Reply:
Sorry you mentioned it somewhere else and I forgot to change it in this isntance